sdk/cosmosdb/cosmos/src/client/SasToken/SasToken.ts

```diff
@@ -0,0 +1,307 @@
+import { CosmosKeyType } from "../../common/constants";
+import { Constants } from "../../common/constants";
+import { encodeUTF8, encodeBase64 } from "../../utils/encode";
+
+export class SasToken implements SasTokenProperties {
+  private AUTH_PREFIX: string = "type=sas&ver=1.0&sig=";
+
+  private SAS_TOKEN_SEPARATOR: string = ";";
+
+  public user: string;
+  public userTag: string;
+  public databaseName: string;
+  public containerName: string;
+  public resourceName: string;
+  public resourcePath: string;
+  public resourceKind: CosmosContainerChildResourceKind;
+  public partitionKeyValueRanges: SasTokenPartitionKeyValueRange[];
+  public startTime: Date = new Date();
+  public expiryTime: Date = new Date();
+  public keyType: number;
+  public controlPlaneReaderScope: number;
+  public controlPlaneWriterScope: number;
+  public dataPlaneReaderScope: number;
+  public dataPlaneWriterScope: number;
+  public cosmosContainerChildResourceKind: CosmosContainerChildResourceKind;
+  public cosmosKeyType = CosmosKeyType.PrimaryMaster;
+  private partitionRanges: string;
+
+  public constructor() {
+    this.user = "";
+    this.userTag = "";
+    this.databaseName = "";
+    this.containerName = "";
+    this.resourceName = "";
+    this.resourcePath = "";
+    this.startTime.setDate(Date.now());
+    this.expiryTime.setDate(Date.now());
+    this.controlPlaneReaderScope = 0;
+    this.controlPlaneWriterScope = 0;
+    this.dataPlaneReaderScope = 0;
+    this.dataPlaneWriterScope = 0;
+    this.keyType = 0;
+    this.cosmosContainerChildResourceKind = CosmosContainerChildResourceKind.Item;
+    this.resourceKind = CosmosContainerChildResourceKind.Item;
+    this.partitionKeyValueRanges = [];
+    this.partitionRanges = "";
+  }
+  public create(user: string, databaseName: string, containerName: string): SasTokenProperties {
+    const token = new SasToken();
+    token.setUser(user);
+    token.setDatabaseName(databaseName);
+    token.setContainerName(containerName);
+    return token;
+  }
+  public sasTokenValueUsingHMAC(key: string): string;
+  public sasTokenValueUsingHMAC(key: string, keyType?: CosmosKeyType) {
+    if (key === "") {
+      throw new Error("key");
+      // return this.SasTokenWithHMACSHA256(key); // check null obj
+    } else if (typeof keyType === "object") {
+      switch (keyType) {
+        case CosmosKeyType.PrimaryMaster:
+          this.keyType = 1;
+        case CosmosKeyType.SecondaryMaster:
+          this.keyType = 2;
+          break;
+        case CosmosKeyType.PrimaryReadOnly:
+          this.keyType = 3;
+          break;
+        case CosmosKeyType.SecondaryReadOnly:
+          this.keyType = 4;
+          break;
+        default:
+          throw new Error("keyType");
+          break;
+      }
+      return this.SasTokenWithHMACSHA256(key);
+    } else {
+      throw new Error("key");
+    }
+  }
+
+  public get DatabaseName(): string {
```

Comment:
Remove all of the getter+setter methods in this class. No need for any of them. Consumers can access value directly: new SaSToken().databaseName